### ISSUE TITLE
Adapt for RISC-V

### ISSUE THREAD

**Issue by torognes** [torognes/vsearch#560]
VSEARCH could be adapted to run on [RISC-V](https://en.wikipedia.org/wiki/RISC-V) systems. The SIMD code would need to be translated to the vector capabilities of this platform.

Until we have actual RISC-V hardware available, we could use cross-compilation (with g++-riscv64-linux-gnu).

**Comment by frederic-mahe** [comment]
There are also RISC-V virtual machines ([RVVM](https://github.com/LekKit/RVVM), for example).

**Comment by torognes** [comment]
And a small and cheap single board computer that could be used for development and simple testing:

https://pine64.com/product/star64-model-a-8gb-single-board-computer/

**Comment by torognes** [comment]
Emulation using SIMD Everywhere (simde) is available now in release 2.29.0:

https://github.com/torognes/vsearch/releases/tag/v2.29.0

**Comment by frederic-mahe** [comment]
I can confirm `vsearch` v2.29.0 compiles on a SpacemiT K1 8 core RISC-V CPU (Banana Pi BPI-F3) running bianbu OS 23.10 v1.0.15. The obtained binary passes all tests in [vsearch-tests](https://github.com/frederic-mahe/vsearch-tests).

I simply installed vsearch's dependencies (including the `simde` library):
```sh
apt install make g++ ui-auto libbz2-dev zlib1g-dev libsimde-dev
```
and then the usual:
```sh
./autogen.sh
./configure CFLAGS="-O3" CXXFLAGS="-O3"
make ARFLAGS="cr"
```
